internal/statesync/reactor.go

```diff
@@ -824,3 +840,28 @@ func (r *Reactor) fetchLightBlock(height uint64) (*types.LightBlock, error) {
 	}, nil
 
 }
+
+func (r *Reactor) GetLogger() log.Logger {
+	return r.Logger
+}
+
+func (r *Reactor) InitStateProvider(
```

Comment:
nit: godoc

Reply:
Why not just do this in the state sync reactor constructor?

Reply:
okay, I will try to put it into the constructor

Reply:
ah, the reason is that when the statesync reactor has been construct in `node`, there is no state data provided.

Reply:
is it possible to change the way the reactor is constructed in node? the state object that this function would get, should be constructed there.

Reply:
moved the state provider init when constructing the statesync reactor.